x-pack/plugins/canvas/server/sample_data/load_sample_data.js

```diff
@@ -20,13 +21,33 @@ export function loadSampleData(server) {
     });
   }
 
+  const sampleDataLinkLabel = i18n.translate('xpack.canvas.sampleDataLinkLabel', {
+    defaultMessage: 'Canvas',
+  });
   server.addSavedObjectsToSampleDataset(
     'ecommerce',
     updateCanvasWorkpadTimestamps(ecommerceSavedObjects)
   );
+  server.addAppLinksToSampleDataset('ecommerce', {
+    path: '/app/canvas#/workpad/workpad-e08b9bdb-ec14-4339-94c4-063bddfd610e',
+    label: sampleDataLinkLabel,
```

Comment:
Why is every label "sampleDataLinkLabel"?

Reply:
`sampleDataLinkLabel ` is a constant that resolves to localized version of `Canvas`

Reply:
That seems kind of magical. I haven't touched a lot of i18n-ified code, is that a normal pattern, where you set some value knowing that the consumer will handle and translate it? Why not do the translation here? 

Also, what happens if other (extenal) plugins provide just some string here, will that string be used like they'd expect?

Reply:
> Why not do the translation here?

Do you mean inside the implementation of `server.addAppLinksToSampleDataset`? You can't do that because the translation id has always to be hardcoded as a string literal inside the `i18n.translate` call (because this is how the existing translations ids are collected to hand them over to the translation agency). But maybe I'm getting you wrong.

Reply:
That's mostly what I meant, but it was a misunderstanding on my part. We're already doing the translation in this file, it's just the same value, so it's assigned to a variable and used in multiple places. I read the changes wrong before, sorry for the noise.